library/core/src/iter/traits/iterator.rs

```diff
@@ -3475,8 +3475,11 @@ pub trait Iterator {
         }
     }
 
-    /// [Lexicographically](Ord#lexicographical-comparison) compares the elements of this [`Iterator`] with those
-    /// of another.
+    /// [Lexicographically](Ord#lexicographical-comparison) compares the [`PartialOrd`] elements of
+    /// this [`Iterator`] with those of another. A [`Option`] type is returned and will either be
+    /// [`Some`] if the types have a total order, or else [`None`]. For example, for floating
```

Comment:
Sorry, I've got more phrasing nitpicks here.

```rust
let it1 = std::iter::repeat(1.0);
let it2 = std::iter::repeat(2.0);
dbg!(it1.partial_cmp(it2));
``` 
[returns `Some(Less)`](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=c55e1a90283e2d2389a089c39cca36a4) here, but that doesn't mean that the *types* have a total order, just that the *values* observed were ordered.

Maybe there's a way to phrase it in terms of if `PartialOrd::partial_cmp` every returns `None` from any of the comparisons done, then this will also return `None`?

Reply:
> Maybe there's a way to phrase it in terms of if PartialOrd::partial_cmp every returns None from any of the comparisons done, then this will also return None?

Wouldn't it be more descriptive to explain that as soon as an orderings can be determined the evaluation stops (similar to how short-circuit evaluation works)?

Reply:
Sure, that sounds good too!